projects/plugins/jetpack/modules/search/instant-search/components/search-app.jsx

```diff
@@ -218,19 +219,25 @@ class SearchApp extends Component {
 	showResults = () => {
 		this.setState( { showResults: true } );
 		this.preventBodyScroll();
+
+		// If we've arrived via a filter or sort only, make sure the search query is set
+		if ( this.props.searchQuery === null ) {
+			this.props.setSearchQuery( '' );
+		}
```

Comment:
What happens if we omit this? Design-wise, I think omitting this search query mutation would make `showResults` easier to understand. 

If we need to set the search query to `''` due to filter or sort changes, I think we might want to implement that in the Redux state with corresponding reducer tests.

Additionally, we might want to add null query value handling to [`generateApiQueryString`](https://github.com/Automattic/jetpack/blob/fix/open-search-results-in-same-tab/projects/plugins/jetpack/modules/search/instant-search/lib/api.js#L188), since the API doesn't seem to handle `null` query values very well:
```js
	if ( query === null ) {
		query = '';
	}
```

Reply:
Resolved in the latest code update.